### PR TITLE
[Bugfix] Check for Invalid Short Name in Team Import

### PR DESCRIPTION
**Pull request by derLalla**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Team names with leading digits lead to internal server errors
Fixes: #2854 

### Description

We are now checking for an invalid short name during the import of a teams json file. 

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise in team mode
4. Add a team via the import via file option
5. The file looks like this:
Invalid file with leading digit in teamName:
```
[{"teamName": "1hallohallo", "username": "artemis_test_user_105"}]
```

File with valid teamName:
```
[{"teamName": "ValidName", "username": "artemis_test_user_105"}]
```
6. Verify that the invalid file throws an error message and the team cannot be imported.

### Invalid team json:
image

#### Valid file:
image
#### Resulting Dialog:
image
#### Imported team:
image

### CODE REVIEW

**Review by airelawaleria — APPROVED** [review]

File: src/test/javascript/spec/component/team/teams-import-from-file-form.component.spec.ts [hunk 1]
```diff
@@ -134,4 +135,19 @@ describe('TeamsImportFromFileFormComponent', () => {
             expect(comp.convertTeams(mockFileStudents)).to.deep.equal(mockFileTeamsConverted);
         });
     });
+
+    describe('Invalid team name throws exception', () => {
+        beforeEach(() => {
+            resetComponent();
```
Comment (FuchsDominik) [inline src/test/javascript/spec/component/team/teams-import-from-file-form.component.spec.ts:141]: Before each for one test can be shortened by moving the `resetComponent();` into the _should throw error_

**Review by FuchsDominik — APPROVED** [review]
Worked well when tested. Only one small remark

**Review by EneaGore — APPROVED** [review]

**Review by tiffca — APPROVED** [review]
Tested successfully

**Review by JohannesWeiss — APPROVED** [review]
Does indeed what it's supposed to do :)

**Review by valeriebucher — APPROVED** [review]
Works as intended!

**Review by krusche — APPROVED** [review]

**Pull request merged**